testing/tests/DevExpress.ui.widgets/tabPanel.tests.js

```diff
@@ -627,6 +629,72 @@ QUnit.module('keyboard navigation', {
     }
 });
 
+
+QUnit.module('Disabled items', {
+    beforeEach() {
+        const items = [
+            { text: 'user', title: 'Personal Data' },
+            { text: 'comment', title: 'Contacts' },
+        ];
+
+        fx.off = true;
+
+        this.$element = $('#tabPanel').dxTabPanel({
+            focusStateEnabled: true,
+            items
+        });
+
+        this.instance = this.$element.dxTabPanel('instance');
+        this.$tabs = this.$element.find(toSelector(TABS_CLASS));
+
+        this.instance.option('items[1].disabled', true);
+        this.instance.focus();
+    },
+    afterEach() {
+        fx.off = false;
+    }
+}, () => {
+    QUnit.test('disabled item can be focused by keyboard', function(assert) {
+        const $disabledItem = $(toSelector(TABS_ITEM_CLASS)).eq(1);
+        const keyboard = keyboardMock(this.$tabs);
+
+        keyboard.press('right');
+
+        assert.strictEqual($disabledItem.hasClass('dx-state-focused'), true, 'disabled item is focused');
+        assert.strictEqual($disabledItem.attr('aria-disabled'), 'true', 'disabled item aria-disabled is correct');
+    });
+
+    QUnit.test('multiview wrapper should have focused class if item is available', function(assert) {
+        const multiViewWrapper = this.$element.find(toSelector(MULTIVIEW_WRAPPER_CLASS));
+        const keyboard = keyboardMock(this.$tabs);
+
+        assert.strictEqual($(multiViewWrapper).hasClass('dx-state-focused'), true, 'focused class set');
+
+        keyboard.press('right');
+        assert.strictEqual($(multiViewWrapper).hasClass('dx-state-focused'), false, 'focused class not set');
+    });
+
+    QUnit.test(`element has ${FOCUS_ON_DISABLED_CLASS} class when disabled item has focus`, function(assert) {
+        const keyboard = keyboardMock(this.$tabs);
+
+        assert.strictEqual($(this.$element).hasClass(FOCUS_ON_DISABLED_CLASS), false, 'class not set');
+
+        keyboard.press('right');
+        assert.strictEqual($(this.$element).hasClass(FOCUS_ON_DISABLED_CLASS), true, 'class set');
+    });
+
+    QUnit.test(`element does not have ${FOCUS_ON_DISABLED_CLASS} class when widget lost focus`, function(assert) {
+        const keyboard = keyboardMock(this.$tabs);
+
+        keyboard.press('right');
+        assert.strictEqual($(this.$element).hasClass(FOCUS_ON_DISABLED_CLASS), true, 'class set');
+
+        this.$element.focusout();
+        assert.strictEqual($(this.$element).hasClass(FOCUS_ON_DISABLED_CLASS), false, 'class not set');
+    });
+});
+
+
```

Comment:
let's add a test for swipe

Reply:
Found a bug while the call.

Reply:
It's ready: [b55c85f86fc93cd1b76a59f40b0c8dc98417d800, de995ccaca3084e389f9c161cb23de8f8696d792, 860b6e80aeaca7745959e0cc398912e12f9f1118].

Reply:
Was fixed in last version.